var/spack/repos/builtin/packages/daos/package.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Daos(SConsPackage):
+    """The Distributed Asynchronous Object Storage (DAOS) is an open-source
+    software-defined object store designed from the ground up for massively
+    distributed Non Volatile Memory (NVM)."""
+
+    homepage = "https://github.com/daos-stack/daos"
+    git = "https://github.com/daos-stack/daos.git"
+    maintainers("hyoklee")
+
+    version("master", branch="master", submodules=True)
+    version("2.2.0", tag="v2.2.0", submodules=True)
+    variant(
+        "debug", default=False, description="Enable debugging info and strict compile warnings"
+    )
+
+    patch("0001-LIBPATH-fix-for-ALT_PREFIX.2.patch", when="@2.2.0:")
+
+    depends_on("argobots@1.1:")
+    depends_on("boost@develop+python", type="build")
+    depends_on("cmocka", type="build")
+    depends_on("go", type="build")
+    depends_on("hwloc")
+    depends_on("isa-l@2.30.0:")
+    depends_on("isa-l-crypto@2.23.0:")
+    depends_on("libfabric@1.15.1:")
+    depends_on("libfuse@3.6.1:")
+    depends_on("libuuid")
+    depends_on("libunwind")
+    depends_on("libyaml")
+    depends_on("mercury@2.2.0:+boostsys")
+    depends_on("mpich")
```

Comment:
@hyoklee mpich is only needed for building certain tests so unless you need to build those tests, you should consider removing the dependency on mpich, it's just going to cause you more issues down the road.